lib/core/services/user_service.dart

```diff
@@ -0,0 +1,84 @@
+import 'package:ata/core/models/auth.dart';
+import 'package:ata/core/models/failure.dart';
+import 'package:ata/util.dart';
+import 'package:dartz/dartz.dart';
+import 'package:intl/intl.dart';
+
+enum AttendanceType { In, Out }
+enum AttendanceStatus { CheckedIn, CheckedOut, NotCheckIn, NotCheckOut }
+
+class UserService {
+  String _localId;
+  String _idToken;
+  String _urlReports = "https://atapp-7720c.firebaseio.com/reports/";
+  UserService(Either<Failure, Auth> auth) {
```

Comment:
UserService(Either<Failure, Auth> auth)  : _auth=auth;  --> should keep a ref to _auth